src/nvim/autocmd.c

```diff
@@ -2568,6 +2568,15 @@ void may_trigger_vim_suspend_resume(bool suspend)
   }
 }
 
+/// Give windows-id in event.data argument
+void do_autocmd_with_winid(event_T event, char *fname, buf_T *buf, Window target_win)
```

Comment:
should this have the same signature as `apply_autocmds` (i.e. include the `force` parameter, so that it's easier to search-and-replace one or the other?

Reply:
@zeertzjq is the objection because of merging vim-patches? What are your thoughts on https://github.com/neovim/neovim/pull/26430#discussion_r1418087057 ?

Reply:
1. It's strange to mark an internal function as "deprecated", as there is no warning for using a "deprecated" function. It's better to just add an extra argument.
2. Makes sense.